### PR TITLE
CAMEL-16301 Allow endpoint to override prefetchCount from component

### PR DESCRIPTION
**Pull request by Kinae**
Endpoint can override the component option prefetchCount to allow more customisation.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: components/camel-spring-rabbitmq/src/main/java/org/apache/camel/component/springrabbit/SpringRabbitMQEndpoint.java [hunk 1]
```diff
@@ -154,6 +154,10 @@
     @UriParam(defaultValue = "false", label = "advanced",
               description = "Sets whether synchronous processing should be strictly used")
     private boolean synchronous;
+    @UriParam(label = "consumer,advanced",
+              description = "Tell the broker how many messages to send in a single request. Often this can be set quite high to improve throughput."
+                            + " This param override the value from the component.")
```
Comment (davsclaus) [inline components/camel-spring-rabbitmq/src/main/java/org/apache/camel/component/springrabbit/SpringRabbitMQEndpoint.java:159]: param -> parameter

But you do not have to tell this, its a common practice for Camel components where options can be configured as default on component level, and override per endpoint if needed.

**Comment by davsclaus** [comment]
You also need to change where the prefectCount is in use today. 

1)
In component set the component level prefetc count on the endpoint createde like the other options are done
endpoint override is done later so you can specify a new value in uri and it will be in use instead of component level

2)
In some listener factory the prefect is in use, its taken from component today, but take it from endpoint

File: components/camel-spring-rabbitmq/src/main/java/org/apache/camel/component/springrabbit/SpringRabbitMQEndpoint.java (hunk 1 above)
Comment (Kinae) [reply components/camel-spring-rabbitmq/src/main/java/org/apache/camel/component/springrabbit/SpringRabbitMQEndpoint.java:159]: Thanks, will remove it

**Comment by Kinae** [comment]
Thank you @davsclaus for the input, learning more and more about Apache Camel each day.

**Review by davsclaus — APPROVED** [review]

**Pull request merged**